app/models/article.rb

```diff
@@ -71,6 +71,17 @@ def initialize(*args)
     end
   end
 
+  def merge_with(other_article_id)
+    other_article = Article.find(other_article_id)
+    Article.transaction do
```

Comment:
Yes! Definitely a transaction is good here. I _think_ you can also use `self.transaction`